benchmark-ph1ex1/convert.py

```diff
@@ -0,0 +1,472 @@
+"""
+This script converts the cross sections in 'OECD-MHTGR350_Simplified.xs'
+to moltres format.
+How to create the XS:
+- Go to main() function
+- Choose one of the 3 options available
+- run 'python convert.py'
+"""
+import numpy as np
+import os
+
+
+def getxs(inFile, index):
+    '''
+    Reads 'OECD-MHTGR350_Simplified.xs' and saves the data into a dictionary
+
+    Parameters:
+    -----------
+    inFile: file
+        path and name of the file
+    index: int
+        material number [1-232]
+    Returns:
+    --------
+    XS: dictionary
+        contains main parameters
+    '''
+
+    with open(inFile, 'r') as i:
+        data = i.readlines()
+    lines = list()
+    for line in data:
+        lines.append(line.split())
+
+    i = int(lines.index(['MATERIAL', str(index)]))
+    XS = {'FLX': [], 'ST': [], 'DIFFCOEF': [], 'NSF': [], 'FISS': [],
+          'CHIT': [], 'SP0': np.zeros((26, 26)), 'SP1': np.zeros((26, 26))}
+    for group in lines[i+1:i+27]:
+        XS['FLX'].append(float(group[1]))
+        XS['ST'].append(float(group[2]))  # \Sigma_t
+        XS['DIFFCOEF'].append(float(group[3]))
+        XS['NSF'].append(float(group[4]))  # \nu \Sigma_f
+        XS['FISS'].append(float(group[5]))  # \Sigma_f
+        XS['CHIT'].append(float(group[6]))
+
+    # XS['SPO'] is a 26 x 26 matrix:
+    # S1-1 S1-2 .... S1-26
+    # S2-1 S2-2 .... S2-26
+    # :::
+    # S26-1 ........ S26-26
+    i += 28
+    for g in range(0, 26):
+        sp0s = int(lines[i+g][0])-1
+        spes = int(lines[i+g][1])-1
+        for gp in range(0, spes-sp0s+1):
+            XS['SP0'][gp+sp0s, g] = float(lines[i+g+26][gp])
+
+    for g in range(0, 26):
+        sp0s = int(lines[i+g][2])-1
+        spes = int(lines[i+g][3])-1
+        for gp in range(0, spes-sp0s+1):
+            XS['SP1'][gp+sp0s, g] = float(lines[i+g+26+26][gp])
+
+    if index == 232:
+        # the 0.9278 is a weighing factor for the crontrol rod
+        # it is specified in the benchmark definition
+        XS['ST'] = 0.9278*np.array(XS['ST'])
+        XS['DIFFCOEF'] = 0.9278*np.array(XS['DIFFCOEF'])
+        XS['NSF'] = 0.9278*np.array(XS['NSF'])
+        XS['FISS'] = 0.9278*np.array(XS['FISS'])
+        XS['SP0'] = 0.9278*np.array(XS['SP0'])
+        XS['SP1'] = 0.9278*np.array(XS['SP1'])
+
```

Comment:
Do not hardcode an actual number 6 times. Give it a variable name. Use that name 6 times.